### PR TITLE
Fix augassign target is evaluated twice in JIT

### PR DESCRIPTION
**Pull request by asi1024**
Reproducer:
```py
import cupy
import cupyx

@cupyx.jit.rawkernel(device=True)
def g(x):
    x[0] += 1
    return 1

@cupyx.jit.rawkernel()
def f(x):
    x[g(x)] += 1

x = cupy.zeros((2,))

f[1, 1](x)

print(x)
print(f.cached_code)
```

master:
```cc
[2. 1.]

__device__ int g_1(CArray<double, 1, true, true> x) {
  x[0] = (x[0] + (double)(1));
  return 1;
}
extern "C" __global__ void f(CArray<double, 1, true, true> x) {
  x[g_1(x)] = (x[g_1(x)] + (double)(1));
}
```

This PR:
```cc
[1. 1.]

__device__ int g_1(CArray<double, 1, true, true> x) {
  { double &tmp__ = x[0]; tmp__ = (tmp__ + (double)(1)); }
  return 1;
}
extern "C" __global__ void f(CArray<double, 1, true, true> x) {
  { double &tmp__ = x[g_1(x)]; tmp__ = (tmp__ + (double)(1)); }
}
```

### CODE REVIEW

**Review by kmaehashi — DISMISSED** [review]
LGTM!

File: cupyx/jit/_compile.py [hunk 1]
```diff
@@ -464,13 +464,18 @@ def _transpile_stmt(stmt, is_toplevel, env):
     if isinstance(stmt, ast.AugAssign):
         value = _transpile_expr(stmt.value, env)
         target = _transpile_expr(stmt.target, env)
-        assert isinstance(target, Data)
+        if not isinstance(target, Data):
+            raise TypeError(f'Cannot augassign to {target.code}')
         value = Data.init(value, env)
-        result = _eval_operand(stmt.op, (target, value), env)
+        tmp = Data('tmp__', target.ctype)
```
Comment (kmaehashi) [inline cupyx/jit/_compile.py:470]: Just curious: Are there any other variable names that are reserved by CuPy JIT?

Comment (asi1024) [reply cupyx/jit/_compile.py]: The users might use `tmp__` as a local variable in the python function. Fixed in [1c229fa](https://github.com/cupy/cupy/pull/6844/commits/1c229fa0e9e2f63b39376594d80ce8b71dafcb24).

Comment (kmaehashi) [reply cupyx/jit/_compile.py]: Looks nice!

**Pull request merged**